packages/e2e-tests/specs/keyboard-navigable-blocks.test.js

```diff
@@ -0,0 +1,165 @@
+			const isFocusedLeftAlignmentControl = await page.evaluate( () =>
+				document.activeElement.classList.contains( 'components-toolbar__control' )
+			);
+			await expect( isFocusedLeftAlignmentControl ).toBe( true );
+
+			// Tab to focus on the 'center paragraph alignment' dropdown control
+			await page.keyboard.press( 'Tab' );
+			const isFocusedCenterAlignmentControl = await page.evaluate( () =>
+				document.activeElement.classList.contains( 'components-toolbar__control' )
+			);
+			await expect( isFocusedCenterAlignmentControl ).toBe( true );
+
+			// Tab to focus on the 'right paragraph alignment' dropdown control
+			await page.keyboard.press( 'Tab' );
+			const isFocusedRightAlignmentControl = await page.evaluate( () =>
+				document.activeElement.classList.contains( 'components-toolbar__control' )
+			);
+			await expect( isFocusedRightAlignmentControl ).toBe( true );
+
+			// Tab to focus on the 'Bold' formatting button
+			await page.keyboard.press( 'Tab' );
+			const isFocusedBoldFormattingButton = await page.evaluate( () =>
+				document.activeElement.classList.contains( 'components-toolbar__control' )
+			);
+			await expect( isFocusedBoldFormattingButton ).toBe( true );
+
+			// Tab to focus on the 'Italic' formatting button
+			await page.keyboard.press( 'Tab' );
+			const isFocusedItalicFormattingButton = await page.evaluate( () =>
+				document.activeElement.classList.contains( 'components-toolbar__control' )
+			);
+			await expect( isFocusedItalicFormattingButton ).toBe( true );
+
+			// Tab to focus on the 'Hyperlink' formatting button
+			await page.keyboard.press( 'Tab' );
+			const isFocusedHyperlinkFormattingButton = await page.evaluate( () =>
+				document.activeElement.classList.contains( 'components-toolbar__control' )
+			);
+			await expect( isFocusedHyperlinkFormattingButton ).toBe( true );
+
+			// Tab to focus on the 'Strikethrough' formatting button
+			await page.keyboard.press( 'Tab' );
+			const isFocusedStrikethroughFormattingButton = await page.evaluate( () =>
+				document.activeElement.classList.contains( 'components-toolbar__control' )
+			);
+			await expect( isFocusedStrikethroughFormattingButton ).toBe( true );
+
+			// Tab to focus on the 'More formatting' dropdown toggle
+			await page.keyboard.press( 'Tab' );
+			const isFocusedMoreFormattingDropdown = await page.evaluate( () =>
+				document.activeElement.classList.contains(
+					'editor-block-settings-menu__toggle'
+				)
+			);
+			await expect( isFocusedMoreFormattingDropdown ).toBe( true );
+		};
+
+		// create 3 paragraphs blocks with some content
+		for ( const paragraphBlock of paragraphBlocks ) {
+			await insertBlock( 'Paragraph' );
+			await page.keyboard.type( paragraphBlock );
+			await page.keyboard.press( 'Enter' );
+		}
+
+		await navigateToContentEditorTop();
+
+		for ( const paragraphBlock of paragraphBlocks ) {
+			await tabThroughParagraphBlock( paragraphBlock );
+		}
+
+		await navigateToContentEditorTop();
```

Comment:
Can you explain why does it navigates through those blocks twice? It feels like it should be fine to do it only once. Maybe it would be enough to navigate to content editor top and finish there. I don't see any technical reasons why it would work differently after this step.

Reply:
Sure can. The regression we're testing for here is illustrated in this image:

![a user tabs through the block editor, resets focus to the top of the screen, and tabs through again. The block's toolbars don't receive focus on the second pass.](https://user-images.githubusercontent.com/1682452/48368776-ebd66c00-e6b4-11e8-94d9-ac96a3213392.gif)

And you can find more information about the bug and the patch for it here: https://github.com/WordPress/gutenberg/issues/11773

Reply:
Okey, we need to document it in the code and reference the bug we are guarding against. Something along those lines:
```js
// Repeat the same steps to ensure that there is no change introduced in how focus is handled.
// This prevents the previous regression explained in: https://github.com/WordPress/gutenberg/issues/11773.
```

What do you think?

Reply:
@gziolo would love your thoughts on this approach here. I could find only one other example of a function from `e2e-test-utils` that contained `expect` assertions like the ones I was using in `tabThroughBlockMoverControl` and `tabThroughBlockToolbar`. 

Currently thinking about the block-agnostic test you mentioned living in `keyboard-navigable-blocks.test.js` and the paragraph order tests written in this PR living in a different file

Reply:
So let's merge this PR as it's solid enough and do a follow-up with the work started in your fork. I didn't test it at all but I anticipate that we will only need to add some JSDoc tweaks to ensure that those new public methods get properly documented in the auto-generated docs :)